.github/workflows/code-freeze.yml

```diff
@@ -32,8 +32,18 @@ jobs:
         echo $BRANCH_NAME
         echo $PR_TITLE
 
-        if [[ "$BRANCH_NAME" != $UNFROZEN_PREFIX* ]] && 
-           [[ "$PR_TITLE" != fix:* && "$PR_TITLE" != *"[critical]"* ]]; then
-          echo "Error: You can only merge from branches that start with '$UNFROZEN_PREFIX', or PRs titled with 'fix: ' and containing '[critical]'."
-          exit 1
+        # if it's not a critical fix
+        if ! [[ "$PR_TITLE" == fix\(critical\):* ]]; then
+          # and there's an unfrozen prefix
+          if ! [[ -z $UNFROZEN_PREFIX ]]; then
+            # check if the branch matches unfrozen prefix
+            if [[ "$BRANCH_NAME" != $UNFROZEN_PREFIX* ]]; then
+              echo "Error: You can only merge from branches that start with '$UNFROZEN_PREFIX', or PRs titled with prefix 'fix(critical): '."
```

Comment:
Note that `fix(critical): ` is not a supported prefix. Granted, fixes and unprefixed commits all lead to patch releases, but unsupported prefixes will appear in the changelist in addition to the commit message.

Reply:
Good point. We will update our parser to handle this case

Reply:
@math411 pointed out that this is the established format for adding scope to the prefix ([source](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#:~:text=A%20scope%20may%20be%20provided%20to%20a%20commit%E2%80%99s%20type%2C%20to%20provide%20additional%20contextual%20information%20and%20is%20contained%20within%20parenthesis%2C%20e.g.%2C%20feat(parser)%3A%20add%20ability%20to%20parse%20arrays.))